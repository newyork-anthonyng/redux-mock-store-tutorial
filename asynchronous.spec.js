import {
  fetchTodos,
  addTodos,
  setFetch,
} from './asynchronous';

// set up redux-mock-store
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
const middlewares = [ thunk ];
const mockStore = configureStore(middlewares);

describe('Using redux-mock-store', () => {
  it('should dispatch correct actions when fetching todos', () => {
    const store = mockStore({});

    return store.dispatch(fetchTodos())
      .then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
  });
});

describe('Without redux-mock-store', () => {
  it('should dispatch correct actions when fetching todos', () => {
    const mockDispatch = jest.fn();

    // `result` is a function that accepts a `dispatch` function
    const result = fetchTodos();

    return result(mockDispatch).then(() => {
      expect(mockDispatch.mock.calls).toMatchSnapshot();
    });
  });
});
