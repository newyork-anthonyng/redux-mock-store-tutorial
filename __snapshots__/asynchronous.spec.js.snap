// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Using redux-mock-store should dispatch correct actions when fetching todos 1`] = `
Array [
  Object {
    "isFetching": true,
    "type": "SET_FETCH",
  },
  Object {
    "todos": Array [
      "Walk dog",
      "Learn redux",
    ],
    "type": "ADD_TODOS",
  },
  Object {
    "isFetching": false,
    "type": "SET_FETCH",
  },
]
`;

exports[`Without redux-mock-store should dispatch correct actions when fetching todos 1`] = `
Array [
  Array [
    Object {
      "isFetching": true,
      "type": "SET_FETCH",
    },
  ],
  Array [
    Object {
      "todos": Array [
        "Walk dog",
        "Learn redux",
      ],
      "type": "ADD_TODOS",
    },
  ],
  Array [
    Object {
      "isFetching": false,
      "type": "SET_FETCH",
    },
  ],
]
`;
